pure.zsh

```diff
@@ -677,8 +677,15 @@ prompt_pure_state_setup() {
 	fi
 
 	hostname='%F{$prompt_pure_colors[host]}@%m%f'
-	# Show `username@host` if logged in through SSH.
-	[[ -n $ssh_connection ]] && username='%F{$prompt_pure_colors[user]}%n%f'"$hostname"
+	# Show `username@host` when:
+	# * logged in through SSH
+	# * inside a docker container
+	# * inside an LXC container
+	if [[ -n $ssh_connection ]] ||
+		( [[ -r /proc/1/cgroup ]] && (( $(grep -cE "(lxc|docker)" /proc/1/cgroup ) > 0 )) ) ||
```

Comment:
Can you move the container logic to a separate function that's called here? `prompt_pure_is_inside_container` or something.

Use long-flags for `grep`.

Reply:
Yes, that makes sense.  Updated to use long flags for grep.

Reply:
After I changed this to use long options, mafredi [pointed out](https://github.com/sindresorhus/pure/pull/564#pullrequestreview-491333614) that long flags are not supported on busybox.  So I changed this block to use short flags.  AFAIK, `-q` is universally supported.  Can we merge it as it is now, or should I revert back to long flags?